syntax = "proto3";

package example;

option cc_generic_services = true;

message LoginRequest {
    bytes name = 1;
    bytes pwd = 2;
};

message RegisterRequest {
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
};

message ResultCode {
    int32 errCode = 1;
    bytes errMsg = 2;
};

message LoginResponse {
    ResultCode code = 1;
    bool success = 2;
}

message RegisterResponse {
    int32 errCode = 1;
    bytes errMsg = 2;
};
service UserService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc Register(RegisterRequest) returns (RegisterResponse);
}